src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/DefaultTagHelperCompletionService.cs

```diff
@@ -199,8 +199,16 @@ public override IReadOnlyList<CompletionItem> GetCompletionsAt(SourceSpan locati
                 return elementCompletions;
             }
 
-            if (_htmlFactsService.TryGetAttributeInfo(parent, out containingTagNameToken, out var selectedAttributeName, out attributes) &&
-                attributes.Span.IntersectsWith(location.AbsoluteIndex))
+            if (_htmlFactsService.TryGetAttributeInfo(
+                    parent,
+                    out containingTagNameToken,
```

Comment:
Don't have to do it now, but with this many `out`'s it feels like there might be a need for a `Context` class. Possibly with helper methods to make some of this logic more explicit.

Reply:
Ya if it gets any larger, definitely going to do that.